File: Julia-files/07_distributions.jl
using Distributions
using Plots

# --- Random Sampling from Normal Distribution ---

# 100 samples from Normal(0,1)
x = rand(Normal(0, 1), 100)

histogram(x, bins=10, color=:skyblue, edgecolor=:black,
          xlabel="Value", ylabel="Frequency",
          title="Normal Distribution Samples")


# --- Probability Density Function (PDF) ---

d = Normal(0, 1)
x_vals = -3:0.1:3
y_vals = pdf.(d, x_vals)

plot(x_vals, y_vals, color=:darkblue,
     xlabel="x", ylabel="Density", title="Normal PDF")


# --- Cumulative Distribution Function (CDF) ---

prob = cdf(d, 1.96)
println(prob)   # ≈ 0.975


# --- Quantiles (Inverse CDF) ---

q = quantile(d, 0.975)
println(q)      # ≈ 1.96


# --- Random sample from N(5, 2) ---

samples = rand(Normal(5, 2), 10)
println(samples)


# --- Comparing Normal and t-Distribution ---

t5 = TDist(5)

plot(x_vals, pdf.(d, x_vals), label="Normal", color=:blue)
plot!(x_vals, pdf.(t5, x_vals), label="t(5)", color=:red,
      title="Normal vs t Distribution")

File: Python-files/07_distributions.py
# Random Sampling
import numpy as np
import matplotlib.pyplot as plt

# Generate 100 random samples from normal distribution (mean=0, sd=1)
x = np.random.normal(loc=0, scale=1, size=100)

# Plot histogram
plt.hist(x, bins=10, color='skyblue', edgecolor='black')
plt.xlabel('Value')
plt.ylabel('Frequency')
plt.title('Normal Distribution Samples')
plt.show()

# Probability Density Function (PDF)
from scipy.stats import norm

# Generate values
x = np.arange(-3, 3, 0.1)
y = norm.pdf(x)

# Plot normal density curve
plt.plot(x, y, color='darkblue')
plt.xlabel('x')
plt.ylabel('Density')
plt.title('Normal PDF')
plt.show()

# Cumulative Distribution Function (CDF)
# Probability that a normal variable <= 1.96
from scipy.stats import norm
prob = norm.cdf(1.96)
print(prob)  # ≈ 0.975

# Quantiles and Random Sampling
# Quantile (inverse of CDF)
quantile = norm.ppf(0.975)
print(quantile)  # ≈ 1.96

# 10 random samples from N(5, 2)
samples = np.random.normal(loc=5, scale=2, size=10)
print(samples)

# Comparing Distributions
from scipy.stats import t

x = np.arange(-3, 3, 0.1)
plt.plot(x, norm.pdf(x), label='Normal', color='blue')
plt.plot(x, t.pdf(x, df=5), label='t(5)', color='red')
plt.legend()
plt.title('Normal vs t Distribution')
plt.show()

